pycbc/inference/prior.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2016  Christopher M. Biwer
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides classes and functions for evaluating the prior
+for parameter estimation.
+"""
+
+def no_prior(params):
+    """ Function that returns default value if user does not specify a prior.
+    """
+    return 0
+
```

Comment:
I tested this and it works right now, but I don't think it'll be possible for the priors to just be functions. They priors will need to know the order and names of parameters when they are passed to them, like the waveform generator does. Better that they be classes, in which one of the class methods is .prior, which takes a list of parameters and returns a value. Could you define a BasePrior (or maybe call it ZeroPrior) that has a variable_args attribute, similar to BaseGenerator? Then you would zip these together with the input parameters, like the generate function. The base one could be the no prior version (so, always returns 0), with future priors inheriting from this and replacing with their own prior functions.